lib/hydra/derivatives/processors/active_encode.rb

```diff
@@ -0,0 +1,55 @@
+require 'active_encode'
+
+module Hydra::Derivatives::Processors
+  class ActiveEncodeError < StandardError
+    def initialize(status, source_path, errors = [])
+      msg = "ActiveEncode status was \"#{status}\" for #{source_path}"
+      msg = "#{msg}: #{errors.join(' ; ')}" unless errors.empty?
+      super(msg)
+    end
+  end
+
+  class ActiveEncode < Processor
+    class_attribute :timeout
+    attr_accessor :encode_class
```

Comment:
Other than when it's initialized this attr is only used once, in `#process`. Would it work instead to have `:encode` be an `attr_reader`? Here's what I have in mind:

```ruby
class ActiveEncode < Processor
  # ...
  attr_reader :encode

  def initialize(source_path, directives, opts = {})
    super
    encode_class = opts.delete(:encode_class) || ::ActiveEncode::Base
    @encode = encode_class.create(sourth_path, directives)
  end
  # ...
```

Then you don't need to pass the `encode` object all over the place.

Related: how would you feel about renaming `:encode` to `:encoder` to make clearer what sort of object this is (using a noun rather than a verb)?